packages/calcite-components/src/components/combobox/combobox.tsx

```diff
@@ -1229,15 +1229,19 @@ export class Combobox
     return this.filterText === "" ? this.items : this.items.filter((item) => !item.hidden);
   }
 
-  private updateItems = debounce((): void => {
```

Comment:
shouldn't we still debounce this in some cases? Like the mutation observer case?

Reply:
We could keep an eye out for this. The MO callback kicks in once per batch of mutations, so I don't expect a significant increase in `updateItem` calls.

Reply:
Sounds good. We may want to have some internal doc/guidelines on when we should debounce a function and under what circumstances.